Document static files behavior (#892)
jupyter-book/myst-theme: docs/permalinks.md

You can match an exact domain (e.g. `mystmd.org`) or use a wildcard to match a single subdomain level. Matches will only be for that subdomain level, not deeper ones, and will not match the root domain (e.g. `*.mystmd.org` matches `docs.mystmd.org` but not `a.b.mystmd.org` or `mystmd.org`).

## Linking to static files

How you link a file determines its URL. See the [MyST downloads guide](https://mystmd.org/guide/website-downloads) for the full reference.

:::{list-table} Static file link behavior
:header-rows: 1

* - Link structure
  - URL you get
  - Notes
  - Example
* - `` {download}`path/to/file.csv` ``
  - `/file-<hash>.csv`

>   - Cache-busted; URL changes with the file

Reviewer: ```suggestion
  - Hashed so that URL changes with the contents of the file
```

Reviewer: Cache-busting is technical jargon that I think a lot of people don't know about.

  - {download}`assets/downloads/example-data.csv`
* - `[text](path/to/file.csv)` (resolves to a source file)
  - `/file-<hash>.csv`
  - Same as the download role
  - [example-data.csv](assets/downloads/example-data.csv)
* - `[text](/file.csv)` - `static_files` **file** entry
  - `/file.csv`
  - Stable (parent folders dropped)
  - [/standalone.csv](/standalone.csv)
* - `[text](/folder/file.csv)` - `static_files` **folder** entry
  - `/folder/file.csv`
  - Stable (folder name kept, parent dropped)
  - [/downloads/example-data.csv](/downloads/example-data.csv)
:::

> A stable link only works if its path doesn't match a source file. Declare the file under `static_files` in `myst.yml`, then link to the resulting URL. The live app redirects such unmatched routes to the served file.

Reviewer: ```suggestion
A static link only works if its path doesn't match a source file. Declare the file under `static_files` in `myst.yml`, then link to the resulting URL. The live app redirects such unmatched routes to the served file.
```

You probably did mean "stable" here, but I don't think it's clear from the context what "stable" means.

> A static link only works if its path doesn't match a source file. Declare the file under `static_files` in `myst.yml`, then link to the resulting URL. The live app redirects such unmatched routes to the served file.

Reviewer: ```suggestion
A static link only works if its path doesn't match a source file. Declare the file under `static_files` in `myst.yml`, then link to the resulting URL.
```

I think this is an implementation detail.

> Each entry is copied to the site root by its **basename**, so parent folders are dropped. You do not need to add a `BASE_URL`, just provide a link relative to your site root. This is the configuration powering the examples above:

Reviewer: ```suggestion
Static files are copied to the site root. To link to them, you can provide a link relative to root: `/my_file.pdf` (no need to include `BASE_URL`). Folder structure is preserved, so if `my_folder` is added to `static_files`, you can link to `/my_folder/my_file.pdf`. This is the configuration powering the examples above:
```
